### ISSUE TITLE
Doesn't always reconnect as network comes up

### ISSUE THREAD

**Issue by GoogleCodeExporter** [baluragala/gtalksms#72]
```
What steps will reproduce the problem?
1. Disable 'connect on wifi' and 'disconnect on no wifi' preferences.  Enable 
'Stop when charge disconnects' preference.
2. Disable 3g, enable wifi, connect gtalksms
3. Disconnect wifi
4. Connect wifi

What is the expected output? What do you see instead?
* Step 3 puts gtalksms into a 'waiting' state.  Step 4 has no effect (ie, 
gtalksms remains waiting).  Expect to see gtalksms reconnect.

For some reason, the network receiver checks the preference 
'stopOnPowerDisconnected' - this seems wrong - the preferences relating to the 
power/charger state should not affect the network connectivity management.  
Attaching a bundle.
```

Original issue reported on code.google.com by `skippy.hammond@gmail.com` on 3 Feb 2011 at 6:55

Attachments:
- [reconnect-fix.bundle](https://storage.googleapis.com/google-code-attachments/gtalksms/issue-72/comment-0/reconnect-fix.bundle)

**Comment by GoogleCodeExporter** [comment]
```
Oups in fact it's a stopOnWifiDisconnected... I'll correct this...
```

Original comment by `Florent....@gmail.com` on 3 Feb 2011 at 6:42

**Comment by GoogleCodeExporter** [comment]
```
Is the inverted boolean in the third case needed at all? If it's true, we will 
land in case 2, if its false, we will land in case 3, with or without it. Or am 
I missing something?
```

Original comment by `fschm...@gmail.com` on 3 Feb 2011 at 6:49

**Comment by GoogleCodeExporter** [comment]
```
Yes no need at all.
```

Original comment by `Florent....@gmail.com` on 3 Feb 2011 at 7:00

**Comment by GoogleCodeExporter** [comment]
Original comment by `Florent....@gmail.com` on 3 Feb 2011 at 7:02
- Changed state: **WaitingForRelease**

**Comment by GoogleCodeExporter** [comment]
```
Fixed in v1.9
```

Original comment by `Florent....@gmail.com` on 6 Feb 2011 at 8:27
- Changed state: **Fixed**

**Comment by GoogleCodeExporter** [comment]
Original comment by `fschm...@gmail.com` on 25 Aug 2011 at 7:51
- Changed state: **Deployed**
